### PR TITLE
Errors added to the result

### PR DESCRIPTION
**Pull request by Dkrenon**
Added some errors to be shown on the console if commands are wrong

### CODE REVIEW

**Review by lbdl — CHANGES_REQUESTED** [review]
this all looks correct but where are the tests? test first please !

**Comment by Dkrenon** [comment]
I'm adjusted the Tokeniser test and added the MeatPuppet Test.  Thanks to that test I was able to notice a problem on the processCommandsTokensLogic but this PR solves it

File: cardinal/system/t_meat_puppet_system_test.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+			assert.Equal(t, tt.expectedOut, output)
+			assert.Equal(t, tt.expectedMove, move)
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+// TestHandleAlias tests the handleAlias function
+func TestHandleAlias(t *testing.T) {
```
Comment (lbdl) [inline cardinal/system/t_meat_puppet_system_test.go:65]: yep this is great. however its usually a good idea to split out tests such that we would test for success and fail/err in separate tests and name appropriately. so we would end up with say: TestHandleAliasErr and TestHandleAlias where the ...Err tests for the error state as below

File: cardinal/system/t_meat_puppet_system_test.go [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := handleAliasTest(tt.tokens, tt.playerID)
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+// TestHandleVerb tests the handleVerb function
+func TestHandleVerb(t *testing.T) {
```
Comment (lbdl) [inline cardinal/system/t_meat_puppet_system_test.go:96]: as above split into 2 tests

File: cardinal/system/t_meat_puppet_system_test.go [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			msg := insultMeatTest(tt.cErr, tt.badCmd)
+			assert.Equal(t, tt.expectedMsg, msg)
+		})
+	}
+}
+
+// Process the Commands tokens, this is the function dedicated to it.
+func ProcessCommandsTokensLogicTest(Tokens []string, PlayerID uint32, roomID uint32) (string, bool, uint8) {
```
Comment (lbdl) [inline cardinal/system/t_meat_puppet_system_test.go:170]: Im not really sure whats going on here, the tests should be on the actual system. It seems like the tests are in fact testing tests, in that in this test you call the handleAliasTest and the handleVerbTest.

File: cardinal/system/t_meat_puppet_system_test.go [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+		err = 135 // This is just for showing errors
+	} else if vrb == enums.ActionTypeLook {
+		fmt.Printf("--->HANDLE ALIAS: NOW SHOULD BE GOING TO STUFF FROM LOOK SYSTEM")
+		err = 130 // This is just for showing errors
+	}
+	return err
+}
+
+// Handle if the token is a verb
+func handleVerbTest(tokens []string, roomID uint32, playerID uint32) (output string, err uint8) {
```
Comment (lbdl) [inline cardinal/system/t_meat_puppet_system_test.go:235]: this is not a test its a function to return some stuff to a test so it only tests tests.

File: cardinal/system/t_meat_puppet_system_test.go [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+		resultStr = "---->HANDLE VERB: NOW SHOULD BE GOING TO ACT FROM ACTION SYSTEM"
+		e = 0
+		fmt.Printf("---->HANDLE VERB:resultStr: %s", resultStr)
+	}
+
+	return resultStr, e
+}
+
+// returns a string to be used on the message variable of the transaction/message
+func insultMeatTest(cErr uint8, badCmd string) string {
```
Comment (lbdl) [inline cardinal/system/t_meat_puppet_system_test.go:267]: again this is not a test it returns something to a tests so its testing a test.

**Review by lbdl — CHANGES_REQUESTED** [review]
This is a good start but its kind of wierd, the tests should test the meatpuppetsystem but instead you have tests that test this test file and functions in the test file.

File: cardinal/system/t_meat_puppet_system_test.go (hunk 1 above)
Comment (Dkrenon) [reply cardinal/system/t_meat_puppet_system_test.go:65]: Leaving this file until get response from Argus guys

File: cardinal/system/t_meat_puppet_system_test.go (hunk 2 above)
Comment (Dkrenon) [reply cardinal/system/t_meat_puppet_system_test.go:96]: Leaving this file until get response from Argus guys

File: cardinal/system/t_meat_puppet_system_test.go (hunk 3 above)
Comment (Dkrenon) [reply cardinal/system/t_meat_puppet_system_test.go:170]: Leaving this file until get response from Argus guys

File: cardinal/system/t_meat_puppet_system_test.go (hunk 4 above)
Comment (Dkrenon) [reply cardinal/system/t_meat_puppet_system_test.go:235]: Leaving this file until get response from Argus guys

File: cardinal/system/t_meat_puppet_system_test.go (hunk 5 above)
Comment (Dkrenon) [reply cardinal/system/t_meat_puppet_system_test.go:267]: Leaving this file until get response from Argus guys

**Comment by Dkrenon** [comment]
-Changed the cardinal version from v1.4.1 to v1.3.3 as there was the compatibility error we find in the morning in order to use their test feature
-On thes test file added the success and failure tests for the Process commands tokens: this inits the world and the game setup, creates the player and do the process commands tokens completely up to the look( gets the room description, the object and directional object) for the success test. For the failure it test the max number of tokens. These test should be the Functional Integration Test or FIT.
-I suggest keeping the existing t_system_meatPuppet_test.go file until we get the answer from the argus guys.

**Review by lbdl — APPROVED** [review]
Fine, good work, please refactor the ols stuff when you get a response from the Argus guys, will approve and merge.

**Pull request merged**